pkg/signup/service/signup_service_test.go

```diff
@@ -173,6 +175,57 @@ func (s *TestSignupServiceSuite) TestSignup() {
 	})
 }
 
+func (s *TestSignupServiceSuite) TestSignupFailsWithNotFoundThenOtherError() {
+
+	// given
+	userID, err := uuid.NewV4()
+	require.NoError(s.T(), err)
+
+	rr := httptest.NewRecorder()
+	ctx, _ := gin.CreateTestContext(rr)
+	ctx.Set(context.UsernameKey, "lisasmith")
+	ctx.Set(context.SubKey, userID.String())
+	ctx.Set(context.OriginalSubKey, "original-sub-value")
+	ctx.Set(context.EmailKey, "lsmith@gmail.com")
+	ctx.Set(context.GivenNameKey, "lisa")
+	ctx.Set(context.FamilyNameKey, "smith")
+	ctx.Set(context.CompanyKey, "red hat")
+
+	callCount := 0
+
+	s.FakeUserSignupClient.MockGet = func(id string) (*toolchainv1alpha1.UserSignup, error) {
+		if callCount == 0 {
```

Comment:
Or you could check the `id` param to return 404 or 500. But the count works too. Just not that obvious.